packages/ui/src/components/theme-selector-v2/theme-dialog.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { FC, useEffect, useState } from 'react'
+
+import { Button, ButtonGroup, Dialog, Separator, Text } from '@/components'
+import { cn } from '@/utils/cn'
+
+import { AccentColor, ColorAdjustment, Contrast, Mode, ThemeDialogProps } from './types'
+
+const ThemeDialog: FC<ThemeDialogProps> = ({
+  defaultTheme,
+  theme,
+  open,
+  onOpenChange,
+  onChange,
+  onSave,
+  onCancel,
+  children
+}) => {
+  const [mode, setMode] = useState<Mode>(Mode.Dark)
+  const [contrast, setContrast] = useState<Contrast>(Contrast.Default)
+  const [colorAdjustment, setColorAdjustment] = useState<ColorAdjustment>(ColorAdjustment.Default)
+  const [accentColor, setAccentColor] = useState<AccentColor>(AccentColor.Blue)
```

Comment:
we could potentially just store the Theme object directly in state instead of all of them separately

Reply:
Done in https://github.com/harness/canary/pull/921.